internal/pkg_web/assembler_spec.js

```diff
@@ -14,21 +15,42 @@ describe('assembler', () => {
       fs.mkdirSync('path');
       fs.mkdirSync('path/to');
       fs.writeFileSync('path/to/thing1.txt', 'some content', {encoding: 'utf-8'});
+      fs.writeFileSync(volatilePath, 'TEST_KEY 41561')
     });
 
     it('should copy files', () => {
-        assembler.main([outdir, '--assets', 'path/to/thing1.txt']);
-        expect(fs.readdirSync('output/path/to')).toContain('thing1.txt');
-        expect(fs.readFileSync('output/path/to/thing1.txt', {encoding: 'utf-8'})).toBe('some content');
+      assembler.main([outdir, volatilePath, '{}', '--assets', 'path/to/thing1.txt']);
+      expect(fs.readdirSync('output/path/to')).toContain('thing1.txt');
+      expect(fs.readFileSync('output/path/to/thing1.txt', {
+        encoding: 'utf-8'
+      })).toBe('some content');
     });
 
     it('should strip longest rootdir', () => {
-        assembler.main([outdir, 'path', 'path/to', '--assets', 'path/to/thing1.txt']);
-        expect(fs.readdirSync('output')).toContain('thing1.txt');
+      assembler.main(
+          [outdir, volatilePath, '{}', 'path', 'path/to', '--assets', 'path/to/thing1.txt']);
+      expect(fs.readdirSync('output')).toContain('thing1.txt');
     });
 
     it('should handle nested directories', () => {
-        assembler.main([outdir, 'path', '--assets', 'path/to']);
-        expect(fs.readdirSync('output/to')).toContain('thing1.txt');
+      assembler.main([outdir, volatilePath, '{}', 'path', '--assets', 'path/to']);
+      expect(fs.readdirSync('output/to')).toContain('thing1.txt');
     });
+
+    it('should replace contents with static text', () => {
+      assembler.main([
+        outdir, volatilePath, '{"some content":"some other content"}', '--assets',
+        'path/to/thing1.txt'
+      ]);
+      expect(fs.readFileSync('output/path/to/thing1.txt', {
+        encoding: 'utf-8'
+      })).toBe('some other content');
+    })
+
+
+    it('should replace contents with dynamic text from volatile file', () => {
+      assembler.main(
+          [outdir, volatilePath, '{"content":"{TEST_KEY}"}', '--assets', 'path/to/thing1.txt']);
```

Comment:
we should make the API match with pkg_npm. Over there we don't let you grab arbitrary keys from the version file, instead we know the key that indicates the version and the `replace_with_version` attribute which says what placeholder is used.

I suspect you prefer this more general implementation? but I think it's more important to be consistent.

Reply:
Yes. I think we should permit the users to replace the contents with the arbitrary keys from the version file. It would be more useful this way. Since we have a key check in bundler, the user would get an error if the key doesn't exist in the version file.

Reply:
In addition, this approach is more general and gives you the ability to replace "placeholder texts" with arbitrary keys from the version file. 

For instance, In one of our projects, we are replacing "BUILD_SCM_HASH" with the real hash along with the  "BUILD_SCM_VERSION" key and it is not possible to accomplish the same behavior with the pkg_npm rule.

Reply:
As far as I understand the rule has to obey --stamp which is stored in the node_context helper rule. I should make this work that way.

Reply:
Perhaps provide both APIs? The `replace_with_version` attr on pkg_web would provide a consistent API surface for both rules, and pkg_npm would change to support substitutions from  workspace_status?

Reply:
update, I'm deprecating `replace_with_version` so let's just do it your way :)